File: FurnitureStore/FurnitureStore.Models/ProductReservation/ProductReservation.cs
﻿
using FurnitureStore.Models.ProductReservationItem;
using FurnitureStore.Models.User;

namespace FurnitureStore.Models.ProductReservation
{
    public class ProductReservation
    {
        public long Id { get; set; }
        public DateTime ReservationDate { get; set; }
        public string Notes { get; set; }
        public bool IsApproved { get; set; }
        public string CustomerId { get; set; }
        public List<ProductReservationItem.ProductReservationItem> ProductReservationItems { get; set; }

    }
}

File: FurnitureStore/UI/furniturestore_mobile/lib/models/product_reservation/product_reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'product_reservation.g.dart';

@JsonSerializable()
class ProductReservationModel {
  int? id;
  DateTime? reservationDate;
  bool? isApproved;
  String? notes;
  String? customerId;
  List<int?>? productReservationItemIds;

  ProductReservationModel({
    this.id,
    this.reservationDate,
    this.isApproved,
    this.notes,
    this.customerId,
    this.productReservationItemIds,
  });

  factory ProductReservationModel.fromJson(Map<String, dynamic> json) =>
      _$ProductReservationModelFromJson(json);

  Map<String, dynamic> toJson() => _$ProductReservationModelToJson(this);
}
